### PR TITLE
7386 - Implement Module Nav Component

### LINKED ISSUE #7386: IdsAppMenu: Module Switcher new Component
**Is your feature request related to a problem or use case? Please describe.**
Add a new side bar view to the app menu. This will encourage teams to simplify the navigation experience

https://www.figma.com/proto/bL4EBYtWjgfBsvTVI9seXU/branch/eHul2nKNMdypFLSb55ZwR1/Application-Menu?page-id=3185%3A78&node-id=3185-6970&viewport=-726%2C826%2C0.5&scaling=scale-down&starting-point-node-id=3185%3A6954

Status: Currently does need some final design clarification. New name "Module Switcher".

**Describe the solution you'd like**

- [ ] Add an option to show the side bar view
- [ ] When hamburger is clicked it animates to the "old view"
- [ ] Options to only show the new view / only show the old view
- [ ] Ability to codify which icons appear (only top level will), so possibly just a slot with icon buttons

**Describe alternatives you've considered**

**Additional context**

- [ ] Masthead should be one on the right of the two (maybe)
- [ ] Logo should line up with masthead

### PR DESCRIPTION
**Pull request by EdwardCoyle**
**Explain the _details_ for making this change. What existing problem does the pull request solve?**
This PR implements the new Module Nav component, and makes modifications to some existing components to add new behavior and styles to those components:

- Added 3 new components: `Module Nav`, `Module Nav Switcher`, and `Module Nav Settings`
- Added Module-nav specific style variants for the Accordion, Dropdown, Popupmenu, and Searchfield components.
- Implemented "sections" into the Accordion that can wrap accordion headers into groups, are navigable, filterable, and can house other components.
- Moved Application Menu filter logic into a shared library used by both Application Menu and Module Nav
- In some cases adjusted existing behavior for detecting IDs, navigating between components with the keyboard, initialization, event handling, etc. to account for implementation in the new component.

The contents of this PR can be seen [here](https://7386-add-nav-menu-enterprise.demo.design.infor.com/components/module-nav/example-index.html)

**Related github/jira issue (required)**:
Closes #7386 

**Steps necessary to review your pull request (required)**:
- Open [the main example](https://7386-add-nav-menu-enterprise.demo.design.infor.com/components/module-nav/example-index.html)
- Try all settings at different viewport sizes.  Watch for strange size discrepancies in sections, odd scrolling behavior, etc.
- Make sure states are correct according to the Figma designs (also watch Design Q/A tab on this file)
- Check [Empty state example](https://7386-add-nav-menu-enterprise.demo.design.infor.com/components/module-nav/example-index.html)
- Check [Six level deep example (no icons)](https://7386-add-nav-menu-enterprise.demo.design.infor.com/components/module-nav/example-six-levels.html)
- Check [Six level deep example (with icons)](https://7386-add-nav-menu-enterprise.demo.design.infor.com/components/module-nav/example-six-levels-icons.html)
- Review modified components for any regressions:
  - [Application Menu](https://7386-add-nav-menu-enterprise.demo.design.infor.com/components/applicationmenu/example-index.html)
  - [Accordion](https://7386-add-nav-menu-enterprise.demo.design.infor.com/components/accordion/example-index.html)
  - [Button](https://7386-add-nav-menu-enterprise.demo.design.infor.com/components/button/example-index.html)
  - [Dropdown](https://7386-add-nav-menu-enterprise.demo.design.infor.com/components/dropdown/example-index.html)
  - [Popupmenu](https://7386-add-nav-menu-enterprise.demo.design.infor.com/components/popupmenu/example-index.html)
  - [Searchfield](https://7386-add-nav-menu-enterprise.demo.design.infor.com/components/searchfield/example-index.html)

**Included in this Pull Request**:
- [x] An e2e or functional test for the bug or feature.
- [x] A note to the change log.

### CODE REVIEW

**Review by tmcconechy — COMMENTED** [review]
Looks good i didnt find much. I fixed your lint errors so the test could run.

1. the alignment looks a bit off now
Screenshot 2023-05-26 at 11 42 15 AM
2. in that same screen shot i think the icon should have 8px not round border radius like the other buttons
3. the dropdown looks a bit disconnected here: maybe it needs an arrow? not sure i didnt see this in the design
Screenshot 2023-05-26 at 11 45 26 AM
4. it looks ok in contrast mode but in dark mode i think it needs a dark treatment
5. as noted in the design QA the only point i think i saw unresolved was adding the hide-focus mixin / classes ect
6. And of course the wrapper later (reminder)

File: app/views/components/module-nav/example-index.html [hunk 1]
```diff
@@ -0,0 +1,369 @@
+<style>
+  .test-spacer {
+    height: 2000px;
```
[the remark this answers is not in the document]
Comment (EdwardCoyle) [reply app/views/components/module-nav/example-index.html:3]: No, this shouldn't be built in, just simulates a tall content area.  I just needed a scaffold for demonstrating a scrollable area works.

**Comment by EdwardCoyle** [comment]
Tim and I chatted on other channels about some of these but just catching up on the PR:

1. I think these have been sufficiently tweaked
2. Button design: [comment thread on Figma](https://www.figma.com/file/eHul2nKNMdypFLSb55ZwR1?node-id=3587:66180#457890843)
3. Dropdown disconnect: [comment thread on Figma](https://www.figma.com/file/eHul2nKNMdypFLSb55ZwR1?node-id=3204:61012#457883386) going for this, but I think it's done as intended
4. I'd asked the design team previously about Dark/Contrast theme colors [in this thread](https://www.figma.com/file/eHul2nKNMdypFLSb55ZwR1?node-id=3587:66421#418498787) and was told they aren't ready just yet.
5. The hide-focus behavior in regards to this component is one of the few points left to address that I'm having difficulty resolving.  I could use some help looking at this part.
6. NG wrapper isn't scheduled yet but we have some bits started on NG branches: https://github.com/infor-design/enterprise-ng/issues/1477

**Comment by tmcconechy** [comment]
Merging this to include in the beta ~ @CindyMercadoReyes can QA do the testing during beta (see board)

**Comment by EdwardCoyle** [comment]
@jbrcna thank you for these.  We are trying to partially fix these on #7575 but some may be pushed further.

Note that for the Safari bug:  Safari turns off Tab key navigation everywhere by default for some reason.  If you run through [the steps on this article](https://articulate.com/support/article/How-to-Enable-Tab-Key-Navigation-on-a-Mac) in Safari, you can activate the feature (I had to do this locally and can confirm tab navigation on headers works as expected after doing so)

**Comment by jbrcna** [comment]
thanks I didn't know that in safari but it believe I did something similar before. I'm not certain if this is because of the update.
the tab  key is now working after I did the steps [on the article](https://articulate.com/support/article/How-to-Enable-Tab-Key-Navigation-on-a-Mac)

**Pull request merged**